### PR TITLE
Add severity to CodeClimate export

### PR DESCRIPTION
**Pull request by jmalmari**
As requested in https://github.com/Ericsson/codechecker/issues/3342#issuecomment-859601266

### CODE REVIEW

File: analyzer/codechecker_analyzer/cmd/parse.py [hunk 1]
```diff
@@ -638,7 +638,7 @@ def parse_convert_reports(input_dirs: List[str],
 
     number_of_reports = len(all_reports)
     if out_format == "codeclimate":
-        return codeclimate.convert(all_reports), number_of_reports
+        return codeclimate.convert(all_reports, severity_map), number_of_reports
```
Comment (csordasmarton) [inline analyzer/codechecker_analyzer/cmd/parse.py:641]: Pylint complains that this line is too long:
```
codechecker_analyzer/cmd/parse.py:641:80: E501 line too long (80 > 79 characters)
make[1]: *** [pycodestyle] Error 1
```
Could you please fix this problem?

File: codechecker_common/output/codeclimate.py [hunk 2]
```diff
@@ -22,18 +22,27 @@ def convert(reports: List[Report]) -> Dict:
...
-        codeclimate_reports.append(__to_codeclimate(report))
+        codeclimate_reports.append(__to_codeclimate(report, severity_map))
     return codeclimate_reports
 
+__codeclimate_severity_map = {
+    'UNDEFINED': 'info',
+    'LOW': 'minor',
+    'STYLE': 'minor',
+    'MEDIUM': 'high',
+    'HIGH': 'critical',
```
Comment (csordasmarton) [inline codechecker_common/output/codeclimate.py:33]: We have the following severity levels:
- Critical
- High
- Medium
- Low
- Style
- Unspecified

So I recommend to change this map to this one:
```suggestion
    'CRITICAL': 'critical',
    'HIGH': 'major',
    'MEDIUM': 'minor',
    'LOW': 'minor',
    'STYLE': 'info',
    'UNSPECIFIED': 'info'
```
What do you think?

File: codechecker_common/output/codeclimate.py [hunk 3]
```diff
@@ -22,18 +22,27 @@ def convert(reports: List[Report]) -> Dict:
...
-def __to_codeclimate(report: Report) -> Dict:
+def __to_codeclimate(report: Report, severity_map: Dict[str, str]) -> Dict:
     """Convert a Report to Code Climate format."""
     return {
         "type": "issue",
         "check_name": report.check_name,
         "description": report.description,
         "categories": ["Bug Risk"],
         "fingerprint": report.report_hash,
+        "severity": __codeclimate_severity_map.get(
```
Comment (csordasmarton) [inline codechecker_common/output/codeclimate.py:44]: Some test cases are failed.
- https://github.com/Ericsson/codechecker/blob/master/analyzer/tests/functional/analyze_and_parse/test_analyze_and_parse.py#L310 - Missing severity level.
- Traceback (most recent call last):
  File "/home/runner/work/codechecker/codechecker/web/tests/functional/diff_local_remote/test_diff_local_remote.py", line 465, in test_diff_codeclimate_output
    self.assertTrue(os.path.exists(codeclimate_issues_file))

Could you please fix these test cases?

File: analyzer/codechecker_analyzer/cmd/parse.py (hunk 1 above)
Comment (csordasmarton) [inline analyzer/codechecker_analyzer/cmd/parse.py:641]: Please update the convert function here with the serverity map:
https://github.com/Ericsson/codechecker/blob/9511fd029f53180bfec3d127306f841ba4a1d427/web/client/codechecker_client/cmd_line_client.py#L1231

**Review by csordasmarton — CHANGES_REQUESTED** [review]
Thank you for this patch :blush: I have some comments on it. If these are fixed and test cases are passed we can merge it.

File: codechecker_common/output/codeclimate.py (hunk 2 above)
Comment (jmalmari) [reply codechecker_common/output/codeclimate.py:33]: Sure, thanks. I have no idea where I got the `high` from.

File: analyzer/codechecker_analyzer/cmd/parse.py (hunk 1 above)
Comment (jmalmari) [reply analyzer/codechecker_analyzer/cmd/parse.py]: fixed

File: codechecker_common/output/codeclimate.py (hunk 2 above)
Comment (jmalmari) [reply codechecker_common/output/codeclimate.py]: fixed

File: codechecker_common/output/codeclimate.py (hunk 3 above)
Comment (jmalmari) [reply codechecker_common/output/codeclimate.py:47]: fixed, though I couldn't run test_functional all the way. I don't have db set up for it.

File: analyzer/codechecker_analyzer/cmd/parse.py (hunk 1 above)
Comment (jmalmari) [reply analyzer/codechecker_analyzer/cmd/parse.py]: fixed

**Review by csordasmarton — APPROVED** [review]
LGTM! Thank you very much for your time and your patch. We really appreciate it :blush:

**Pull request merged**